lightning-invoice/src/utils.rs

```diff
@@ -237,7 +240,13 @@ where L::Target: Logger {
 		&self, payer: &PublicKey, params: &RouteParameters, _payment_hash: &PaymentHash,
 		first_hops: Option<&[&ChannelDetails]>, scorer: &S
 	) -> Result<Route, LightningError> {
-		find_route(payer, params, &*self.network_graph, first_hops, &*self.logger, scorer)
+		let random_seed_bytes: [u8; 32];
+		{
+			let mut locked_random_seed_bytes = self.random_seed_bytes.lock().unwrap();
+			*locked_random_seed_bytes = sha256::Hash::hash(&*locked_random_seed_bytes).into_inner();
+			random_seed_bytes = *locked_random_seed_bytes;
+		}
```

Comment:
Note this could also be written as:

```rust
let random_seed_bytes = {
    // ...
   *locked_random_seed_bytes
};
```

Reply:
Indeed much cleaner. Done with https://github.com/lightningdevkit/rust-lightning/pull/1286/commits/7c5ee057822dcb690c542e6aa0ea87fb4882a266.